### ISSUE TITLE
Need automated testing for Linux distros other than Ubuntu

### ISSUE THREAD

**Issue by delan** [servo/servo#30592]
According to [our docs](https://github.com/servo/servo/wiki/Building#debian-based-distributions), we support several Linux distros, but we only test Ubuntu 20.04 on CI:

* Debian et al (+ mach bootstrap)
* Fedora et al (+ mach bootstrap)
* Void (+ mach bootstrap)
* Arch
* openSUSE
* Gentoo
* NixOS

We should create builds for the other distros, or if that’s infeasible, we should consider requiring Nix on all Linux distros.

**Comment by mrobinson** [comment]
I'm not sure that requiring nix on all platforms is the right way to go, but maybe we can offer support for top five and then for all others require compilation through nix or some other standard environment. I assume that even for nix there are different versions that we need to care about?

**Comment by delan** [comment]
Not really! While the nixpkgs tree is rolling release (with stable branches for NixOS), projects that use Nix can pin their deps indefinitely and use deps as old or new as needed, irrespective of what the user has installed.

In theory, we can give everyone on Linux what Servo [already](https://github.com/servo/servo/blob/22f877c6ab1561e19ca26f4f1e1f2ee5312455f9/etc/shell.nix) [has](https://github.com/servo/servo/blob/22f877c6ab1561e19ca26f4f1e1f2ee5312455f9/mach#L34-L44) on NixOS: one unified Linux build environment, seamless to users other than them having to install Nix. mach would work without installing anything else, and we would eliminate the entire class of breakages caused by distro-packaged deps being too old or new.

We would still want to support and test running prebuilt nightlies without Nix on a variety of distros, but I think [our runtime deps](https://github.com/servo/servo/blob/22f877c6ab1561e19ca26f4f1e1f2ee5312455f9/README.md#runtime-dependencies) are a much easier list to maintain.
